sycl/source/detail/queue_impl.cpp

```diff
@@ -86,6 +91,79 @@ void queue_impl::addEvent(event Event) {
   MEvents.push_back(std::move(Event));
 }
 
+void queue_impl::wait(const cl::sycl::detail::code_location &CodeLoc) {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  xpti::trace_event_data_t *TraceEvent = nullptr;
+  std::string Name;
+  int32_t StreamID = xptiRegisterStream(SYCL_STREAM_NAME);
+  if (xptiTraceEnabled()) {
+    xpti::payload_t p;
+    bool hasSourceInfo = false;
+    // We try to create a unique string for the wait() call
+    // by combining it with the queue address
+    xpti::utils::StringHelper NG;
+    Name = NG.nameWithAddress("queue.wait", (void *)this);
+
+    std::string Check(CodeLoc.fileName());
+    if (Check.find("unknown") == std::string::npos) {
```

Comment:
Maybe checking for column == 0 instead?

Reply:
Unfortunately, column == 0 at all times, but I think there is a desire to add a builtin to get columns in GCC and clang?

Reply:
Have changed logic to check for empty string now - this should make it lighter weight than before.

Reply:
Thanks!